<launch>
    <arg name="cmd_src" default="circle"/>
    <arg name="robot" default="nusim"/>
    <arg name="use_rviz" default="true"/>

    <!-- cmd_src argument -->
    <node pkg="nuturtle_control" exec="circle_node" name="circle_node" if="$(eval '\'$(var cmd_src)\' == \'circle\'')" >
        <param name="frequency" value="100.0"/>
    </node>

    <node pkg="turtlebot3_teleop" exec="teleop_twist_keyboard" if="$(eval '\'$(var cmd_src)\' == \'teleop\'')" >
    </node>

    <!-- robot argument -->
    <group if="$(eval '\'$(var robot)\' == \'nusim\'')" >

        <node pkg="nusim" exec="nusim_node" name="nusim_node" output="screen" >
            <param from="$(find-pkg-share nusim)/config/basic_world.yaml"/>
            <remap from="wheel_cmd" to="red/wheel_cmd" />
            <remap from="joint_states" to="red/joint_states" />
        </node>

        <include file="$(find-pkg-share nuturtle_description)/launch/load_one.launch.py" >
            <arg name="use_rviz" value="false" />
            <arg name="color" value="red" />
        </include>

        <node pkg="nuturtle_control" exec="odom_node" >
            <param name="body_id" value="blue/base_footprint" />
            <param name="odom_id" value="odom" />
            <param name="wheel_left" value="wheel_left" />
            <param name="wheel_right" value="wheel_right" />
            <param from="$(find-pkg-share nuturtle_description)/config/diff_params.yaml"/>
            <remap from="joint_states" to="blue/joint_states" />
        </node>

        <node pkg="nuturtle_control" exec="turtle_control_node" >
            <param from="$(find-pkg-share nuturtle_description)/config/diff_params.yaml"/>
            <remap from="wheel_cmd" to="red/wheel_cmd" />
            <remap from="joint_states" to="blue/joint_states" />
        </node>

        <node pkg="tf2_ros" exec="static_transform_publisher" name="nusim_world_to_odom" args="--frame-id nusim/world --child-frame-id odom" />

    </group>

    <group if="$(eval '\'$(var use_rviz)\' == \'true\'')" >
        <node pkg="rviz2" exec="rviz2" name="rviz2" args="-d $(find-pkg-share nuturtle_control)/config/start_robot.rviz -f nusim/world">
        </node>
    </group>

    <include file="$(find-pkg-share nuturtle_description)/launch/load_one.launch.py" >
        <arg name="use_rviz" value="false" />
        <arg name="color" value="blue" />
    </include>

</launch>
